### PR TITLE
feat(signpost): initial design guidance

### PR DESCRIPTION
**Pull request by naga-zone**
Added initial guidance for signpost.
[CDE-2375](https://vmw-jira.broadcom.net/browse/CDE-2375) - Create MD file for Signpost Component.

### CODE REVIEW

File: 1027-signpost-design-guidance.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+[Signposts](https://clarity.design/documentation/signpost) are displayed in a popover dialog box and contain contextual help or information.
+
+## Guidance
+
+- **Must** use as an alternative to [tooltip](https://clarity.design/documentation/tooltip) when you have more content.
+- **Must** use when the information presented needs to stay in view until the user interacts with another element.
+- **Must** use when you want to include a header, an image, or text/image links in the content, as [tooltip](https://clarity.design/documentation/tooltip) does not support it.
+- **Must** use the allowed dimensions for the dialog. Width X Height format (216px minimum X 84px minimum) or (360px maximum X 504px maximum).
+- **Must not** insert any actionable components, such as [buttons](https://clarity.design/documentation/button), in the dialog box. You can use text or image links.
+- **Should** be paired with a trigger element, such as an icon or button link.
```
Comment (kevinbuhmann) [inline 1027-signpost-design-guidance.md:19]:
```suggestion
- **Must** be paired with a trigger element, such as an icon or button link.
```

How else would it work?

File: 1027-signpost-design-guidance.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+updated: 2024-10-22
+---
+
+# Signpost Design Guidance
+
+[Signposts](https://clarity.design/documentation/signpost) are displayed in a popover dialog box and contain contextual help or information.
+
+## Guidance
+
+- **Must** use as an alternative to [tooltip](https://clarity.design/documentation/tooltip) when you have more content.
```
Comment (jbusta5) [inline 1027-signpost-design-guidance.md:14]:
```suggestion
- Design teams **must** use signposts as an alternative to tooltips when additional content is necessary. This allows for more detailed information to be provided without overwhelming the user interface.
```

File: 1027-signpost-design-guidance.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+- **Must** use when the information presented needs to stay in view until the user interacts with another element.
```
Comment (jbusta5) [inline 1027-signpost-design-guidance.md:15]:
```suggestion
- Design teams **must** use signposts when the information needs to remain visible until the user interacts with another element. This ensures that users have access to important contextual information while navigating the interface.
```

File: 1027-signpost-design-guidance.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+- **Must** use when you want to include a header, an image, or text/image links in the content, as [tooltip](https://clarity.design/documentation/tooltip) does not support non-text or interactive content.
```
Comment (jbusta5) [inline 1027-signpost-design-guidance.md:16]:
```suggestion
- Design teams **must** use signposts when including a header, image, or text/image links in the content, as tooltips do not support non-text or interactive elements. This allows for a richer presentation of information.
```

File: 1027-signpost-design-guidance.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+- **Must** use the allowed dimensions for the dialog. Width X Height format (216px minimum X 84px minimum) or (360px maximum X 504px maximum).
```
Comment (jbusta5) [inline 1027-signpost-design-guidance.md:17]:
```suggestion
- Design teams **must** use the specified dimensions for the dialog, adhering to the format of Width X Height (216px minimum X 84px minimum) or (360px maximum X 504px maximum). This ensures consistency and optimal usability across the interface.
```

File: 1027-signpost-design-guidance.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+- **Must** be paired with a trigger element, such as a button or icon button.
```
Comment (jbusta5) [inline 1027-signpost-design-guidance.md:18]:
```suggestion
- Design teams **must** pair signposts with a trigger element, such as a button or icon button. This establishes a clear interaction pathway for users, making it easy for them to access contextual information when needed.
```

File: 1027-signpost-design-guidance.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+- **Must not** insert any actionable components, such as [buttons](https://clarity.design/documentation/button), in the dialog box. You can use text or image links.
```
Comment (jbusta5) [inline 1027-signpost-design-guidance.md:19]:
```suggestion
- Design teams **must not** insert any actionable components, such as buttons, in the dialog box. Instead, text or image links can be used to maintain clarity and prevent confusion about the dialog's purpose.
```

File: 1027-signpost-design-guidance.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+- **Should not** be so large that it dominates the screen. Make sure that the popover doesn't obstruct other important information on the screen.
```
Comment (jbusta5) [inline 1027-signpost-design-guidance.md:20]:
```suggestion
- Design teams **should not** allow signposts to become so large that they dominate the screen. It’s important to ensure that the popover does not obstruct other critical information, maintaining a clear user interface.
```

File: 1027-signpost-design-guidance.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+- **Should** open only one signpost at a time to keep the interface uncluttered.
```
Comment (jbusta5) [inline 1027-signpost-design-guidance.md:21]:
```suggestion
- Design teams **should** ensure that only one signpost opens at a time to keep the interface uncluttered. This approach helps users focus on the information presented without being overwhelmed by multiple popovers.
```

File: 1027-signpost-design-guidance.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+- **May** include a vertical scrollbar (browser default), if the content exceeds the maximum height of the dialog.
```
Comment (jbusta5) [inline 1027-signpost-design-guidance.md:22]:
```suggestion
- Design teams **may** include a vertical scrollbar (browser default) if the content exceeds the maximum height of the dialog. This ensures that users can still access all information without compromising the dialog's design integrity.
```

**Review by kevinbuhmann — APPROVED** [review]

**Pull request merged**